components/bl-category-chart/src/index.js

```diff
@@ -30,6 +30,12 @@ export default function CategoryChartComponent({ component, elRef }) {
 
   const styles = { ...style, width, height };
 
+  useEffect(() => {
+    if (!chartLegendVisibility) {
+      Chart.defaults.plugins.legend.display = false;
+    }
+  }, [chartLegendVisibility]);
```

Comment:
If `chartLegendVisibility` is not dynamic

It could be like

```
  useEffect(() => {
    Chart.defaults.plugins.legend.display = chartLegendVisibility;
  }, []);
  ```